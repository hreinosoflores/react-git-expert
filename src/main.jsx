import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from './GifExpertApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    //Console.log se ejecuta dos veces
    //<React.StrictMode>
    <GifExpertApp />
    //</React.StrictMode>
);
