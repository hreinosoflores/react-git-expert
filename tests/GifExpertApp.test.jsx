import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { GifExpertApp } from '../src/GifExpertApp';


// eslint-disable-next-line no-undef
describe('pruebas en GifExpertApp', () => {
    // eslint-disable-next-line no-undef
    test('should first', () => {        
        const { container } = render(<GifExpertApp />);
        screen.debug();
        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: { value: 'Saitama' } });
        // eslint-disable-next-line no-undef
        expect(container).toMatchSnapshot();
    });
});