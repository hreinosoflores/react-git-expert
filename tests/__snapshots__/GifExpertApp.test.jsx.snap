// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp should first 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value="Saitama"
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
